### ISSUE TITLE
Oxidized backups of OneAccess device configuration broken

### ISSUE THREAD

**Issue by cyberbohu** [ytti/oxidized#1996]
**Environnement:**
Ubuntu 18.4
oxidized-0.27.0

I tried to backup OneAccess One20 / One50 (ONEOS6-MONO-V5.2R2E8)
I checked oneos.rb
First of all, in ONEOS6, the prompt is ">". In oneos.rb line prompt is "prompt /^([\w.@()-]+#\s?)$/", so I added ">" like that: "prompt /^([\w.@-]+[#>]\s?)$/"
when I run oxidized, my backup fail on my oneaccess, apparently, oxidized put correctly the command "term len 0" (this command works on my oneaccess) but don't press "enter" to execute this command
Any help? 

**Debug:**
---
D, [2019-12-24T10:05:53.753502 #3361] DEBUG -- : lib/oxidized/job.rb: Starting fetching process for 10.252.4.36 at 2019-12-24 09:05:53 UTC
D, [2019-12-24T10:05:53.753887 #3361] DEBUG -- : lib/oxidized/input/ssh.rb: Connecting to 10.252.4.36
D, [2019-12-24T10:05:53.754071 #3361] DEBUG -- : AUTH METHODS::["password", "keyboard-interactive"]
D, [2019-12-24T10:05:53.755307 #3361] DEBUG -- net.ssh.transport.session[3ffb600ff398]: establishing connection to 10.252.4.36:22
D, [2019-12-24T10:05:53.764679 #3361] DEBUG -- net.ssh.transport.session[3ffb600ff398]: connection established
I, [2019-12-24T10:05:53.764833 #3361]  INFO -- net.ssh.transport.server_version[3ffb600fed08]: negotiating protocol version
D, [2019-12-24T10:05:53.764895 #3361] DEBUG -- net.ssh.transport.server_version[3ffb600fed08]: local is 'SSH-2.0-Ruby/Net::SSH_5.2.0 x86_64-linux-gnu'
D, [2019-12-24T10:05:53.826608 #3361] DEBUG -- net.ssh.transport.server_version[3ffb600fed08]: remote is `SSH-2.0-OneSSH_3.9p1'
I, [2019-12-24T10:05:53.826822 #3361]  INFO -- net.ssh.transport.algorithms[3ffb600fe4ac]: sending KEXINIT
D, [2019-12-24T10:05:53.827117 #3361] DEBUG -- socket[3ffb600ff17c]: queueing packet nr 0 type 20 len 1324
D, [2019-12-24T10:05:53.827244 #3361] DEBUG -- socket[3ffb600ff17c]: sent 1328 bytes
D, [2019-12-24T10:05:53.851129 #3361] DEBUG -- socket[3ffb600ff17c]: read 440 bytes
D, [2019-12-24T10:05:53.851291 #3361] DEBUG -- socket[3ffb600ff17c]: received packet nr 0 type 20 len 436
I, [2019-12-24T10:05:53.851376 #3361]  INFO -- net.ssh.transport.algorithms[3ffb600fe4ac]: got KEXINIT from server
I, [2019-12-24T10:05:53.851538 #3361]  INFO -- net.ssh.transport.algorithms[3ffb600fe4ac]: negotiating algorithms
D, [2019-12-24T10:05:53.851687 #3361] DEBUG -- net.ssh.transport.algorithms[3ffb600fe4ac]: negotiated:
* kex: diffie-hellman-group14-sha1
* host_key: ssh-rsa
* encryption_server: aes256-ctr
* encryption_client: aes256-ctr
* hmac_client: hmac-sha2-512
* hmac_server: hmac-sha2-512
* compression_client: none
* compression_server: none
* language_client:
* language_server:
D, [2019-12-24T10:05:53.851751 #3361] DEBUG -- net.ssh.transport.algorithms[3ffb600fe4ac]: exchanging keys
D, [2019-12-24T10:05:53.867564 #3361] DEBUG -- socket[3ffb600ff17c]: queueing packet nr 1 type 30 len 268
D, [2019-12-24T10:05:53.867692 #3361] DEBUG -- socket[3ffb600ff17c]: sent 272 bytes
D, [2019-12-24T10:05:54.753626 #3361] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2019-12-24T10:05:55.513100 #3361] DEBUG -- socket[3ffb600ff17c]: read 592 bytes
D, [2019-12-24T10:05:55.513525 #3361] DEBUG -- socket[3ffb600ff17c]: received packet nr 1 type 31 len 572
D, [2019-12-24T10:05:55.516951 #3361] DEBUG -- socket[3ffb600ff17c]: queueing packet nr 2 type 21 len 20
D, [2019-12-24T10:05:55.517320 #3361] DEBUG -- socket[3ffb600ff17c]: sent 24 bytes
D, [2019-12-24T10:05:55.517702 #3361] DEBUG -- socket[3ffb600ff17c]: received packet nr 2 type 21 len 12
D, [2019-12-24T10:05:55.518621 #3361] DEBUG -- net.ssh.authentication.session[3ffb600572ec]: beginning authentication of 'MYUSERID'
D, [2019-12-24T10:05:55.519070 #3361] DEBUG -- socket[3ffb600ff17c]: queueing packet nr 3 type 5 len 28
D, [2019-12-24T10:05:55.519399 #3361] DEBUG -- socket[3ffb600ff17c]: sent 96 bytes
D, [2019-12-24T10:05:55.620834 #3361] DEBUG -- socket[3ffb600ff17c]: read 96 bytes
D, [2019-12-24T10:05:55.621267 #3361] DEBUG -- socket[3ffb600ff17c]: received packet nr 3 type 6 len 28
D, [2019-12-24T10:05:55.621534 #3361] DEBUG -- net.ssh.authentication.session[3ffb600572ec]: trying password
D, [2019-12-24T10:05:55.621824 #3361] DEBUG -- socket[3ffb600ff17c]: queueing packet nr 4 type 50 len 60
D, [2019-12-24T10:05:55.621990 #3361] DEBUG -- socket[3ffb600ff17c]: sent 128 bytes
D, [2019-12-24T10:05:55.753985 #3361] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2019-12-24T10:05:56.323459 #3361] DEBUG -- socket[3ffb600ff17c]: read 80 bytes
D, [2019-12-24T10:05:56.323670 #3361] DEBUG -- socket[3ffb600ff17c]: received packet nr 4 type 52 len 12
D, [2019-12-24T10:05:56.323832 #3361] DEBUG -- net.ssh.authentication.methods.password[3ffb60056144]: password succeeded
D, [2019-12-24T10:05:56.324100 #3361] DEBUG -- socket[3ffb600ff17c]: queueing packet nr 5 type 90 len 44
D, [2019-12-24T10:05:56.324232 #3361] DEBUG -- : lib/oxidized/input/ssh.rb: expecting [/^([\w.@-]+[#>]\s?)$/] at 10.252.4.36
D, [2019-12-24T10:05:56.525026 #3361] DEBUG -- socket[3ffb600ff17c]: sent 112 bytes
D, [2019-12-24T10:05:56.625369 #3361] DEBUG -- socket[3ffb600ff17c]: read 96 bytes
D, [2019-12-24T10:05:56.625569 #3361] DEBUG -- socket[3ffb600ff17c]: received packet nr 5 type 91 len 28
I, [2019-12-24T10:05:56.625716 #3361]  INFO -- net.ssh.connection.session[3ffb60055078]: channel_open_confirmation: 0 0 0 32768
I, [2019-12-24T10:05:56.625891 #3361]  INFO -- net.ssh.connection.channel[3ffb60054f4c]: sending channel request "env"
D, [2019-12-24T10:05:56.626087 #3361] DEBUG -- socket[3ffb600ff17c]: queueing packet nr 6 type 98 len 44
I, [2019-12-24T10:05:56.626229 #3361]  INFO -- net.ssh.connection.channel[3ffb60054f4c]: sending channel request "pty-req"
D, [2019-12-24T10:05:56.626423 #3361] DEBUG -- socket[3ffb600ff17c]: queueing packet nr 7 type 98 len 60
D, [2019-12-24T10:05:56.726820 #3361] DEBUG -- socket[3ffb600ff17c]: sent 240 bytes
D, [2019-12-24T10:05:56.754266 #3361] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2019-12-24T10:05:56.827176 #3361] DEBUG -- socket[3ffb600ff17c]: read 176 bytes
D, [2019-12-24T10:05:56.827375 #3361] DEBUG -- socket[3ffb600ff17c]: received packet nr 6 type 93 len 28
I, [2019-12-24T10:05:56.827482 #3361]  INFO -- net.ssh.connection.session[3ffb60055078]: channel_window_adjust: 0 +131072
D, [2019-12-24T10:05:56.827606 #3361] DEBUG -- socket[3ffb600ff17c]: received packet nr 7 type 99 len 12
I, [2019-12-24T10:05:56.827694 #3361]  INFO -- net.ssh.connection.session[3ffb60055078]: channel_success: 0
I, [2019-12-24T10:05:56.827767 #3361]  INFO -- net.ssh.connection.channel[3ffb60054f4c]: sending channel request "shell"
D, [2019-12-24T10:05:56.827914 #3361] DEBUG -- socket[3ffb600ff17c]: queueing packet nr 8 type 98 len 28
D, [2019-12-24T10:05:56.928318 #3361] DEBUG -- socket[3ffb600ff17c]: sent 96 bytes
D, [2019-12-24T10:05:57.028685 #3361] DEBUG -- socket[3ffb600ff17c]: read 192 bytes
D, [2019-12-24T10:05:57.028870 #3361] DEBUG -- socket[3ffb600ff17c]: received packet nr 8 type 99 len 12
I, [2019-12-24T10:05:57.029059 #3361]  INFO -- net.ssh.connection.session[3ffb60055078]: channel_success: 0
D, [2019-12-24T10:05:57.029222 #3361] DEBUG -- socket[3ffb600ff17c]: received packet nr 9 type 94 len 44
I, [2019-12-24T10:05:57.029309 #3361]  INFO -- net.ssh.connection.session[3ffb60055078]: channel_data: 0 29b
D, [2019-12-24T10:05:57.129856 #3361] DEBUG -- : lib/oxidized/input/cli.rb: Running post_login commands at 10.252.4.36
D, [2019-12-24T10:05:57.129963 #3361] DEBUG -- : lib/oxidized/input/cli.rb: Running post_login command: "term len 0", block: nil at 10.252.4.36
D, [2019-12-24T10:05:57.130021 #3361] DEBUG -- : lib/oxidized/input/ssh.rb term len 0 @ 10.252.4.36 with expect: /^([\w.@-]+[#>]\s?)$/
D, [2019-12-24T10:05:57.130191 #3361] DEBUG -- socket[3ffb600ff17c]: queueing packet nr 9 type 94 len 28
D, [2019-12-24T10:05:57.130302 #3361] DEBUG -- : lib/oxidized/input/ssh.rb: expecting [/^([\w.@-]+[#>]\s?)$/] at 10.252.4.36
D, [2019-12-24T10:05:57.330966 #3361] DEBUG -- socket[3ffb600ff17c]: sent 96 bytes
D, [2019-12-24T10:05:57.431303 #3361] DEBUG -- socket[3ffb600ff17c]: read 96 bytes
D, [2019-12-24T10:05:57.431491 #3361] DEBUG -- socket[3ffb600ff17c]: received packet nr 10 type 94 len 28
I, [2019-12-24T10:05:57.431594 #3361]  INFO -- net.ssh.connection.session[3ffb60055078]: channel_data: 0 10b
D, [2019-12-24T10:05:57.754552 #3361] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2019-12-24T10:05:58.754826 #3361] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2019-12-24T10:05:59.755141 #3361] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel

**Config file:**
---
username: MYUSERNAME
password: password
model: junos
resolve_dns: false
interval: 3600
use_syslog: false
debug: false
threads: 30
timeout: 100
retries: 3
prompt: !ruby/regexp /^([\w.@-]+[#>]\s?)$/
rest: 10.1.95.34:8000/oxidized
next_adds_job: false
groups: {}
models:
  junos:
username: MYUSERNAME
password: test
    vars:
      auth_methods:
      - publickey
      ssh_keys: "tild/.ssh/id_rsa"
  oneos:
    username: MYUSERNAME
    password: MYPASSWORD
    vars:
      auth_methods: [ "password", "keyboard-interactive" ]
      ssh_no_keepalive: true
      ssh_no_exec: true
pid: "/home/everko/.config/oxidized/pid"
crash:
  directory: "/home/everko/.config/oxidized/crashes"
  hostnames: false
stats:
  history_size: 10
input:
  default: ssh
  debug: true
  ssh:
    secure: false
  ftp:
    passive: true
  utf8_encoded: true
output:
  default: file
  file:
    directory: "/home/everko/.config/oxidized/configs"
source:
  default: csv
  file: "tield/.config/oxidized/router.db"
  delimiter: !ruby/regexp /:/
  map:
    name: 0
    model: 1
  csv:
    file: "/home/everko/.config/oxidized/router.db"
    delimiter: !ruby/regexp /:/
    map:
      name: 0
      model: 1
    gpg: false
model_map:
  juniper: junos
  cisco: ios
  onesaccess: oneos

**Comment by cyberbohu** [comment]
I actually use the regex /^([\w.@-]+[#>]\s?)$/
my prompt, when I'm log to my device is:
STMAC-036_P_NATION_1-L3-01>
It match, so I don't understand whats the problem

**Comment by cyberbohu** [comment]
I changed some line from oneos.rb
exemple: 
cmd 'show running-config' do |cfg|
to
cmd "show running-config\r" do |cfg|
post_login "term len 0\r"
pre_logout "exit\r"

and finally my backup work

**Comment by sgalliani-efs** [comment]
Hello Cyber i was wondering if you could help me as i'm having the same problem as it's giving me a headache,

I modified somethings like you did but i'm still getting a telnet failed and other weird errors, this is my oneos : 

[11:53] Stefano GALLIANI
    a"
D, [2020-08-25T09:53:32.220353 #2297] DEBUG -- : lib/oxidized/node.rb: Oxidized::Telnet failed for RENNES_Nominal
D, [2020-08-25T09:53:32.220394 #2297] DEBUG -- : lib/oxidized/job.rb: Config fetched for RENNES_Nominal at 2020-08-25 09:53:32 UTC
W, [2020-08-25T09:53:32.891763 #2297]  WARN -- : /RENNES_Nominal status no_connection, retry attempt 1
D, [2020-08-25T09:53:32.891862 #2297] DEBUG -- : lib/oxidized/worker.rb: Jobs running: 0 of 1 - ended: 0 of 6
D, [2020-08-25T09:53:32.892700 #2297] DEBUG -- : lib/oxidized/worker.rb: Added /RENNES_Nominal to the job queue
D, [2020-08-25T09:53:32.892739 #2297] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2020-08-25T09:53:32.892969 #2297] DEBUG -- : lib/oxidized/job.rb: Starting fetching process for RENNES_Nominal at 2020-08-25 09:53:32 UTC
D, [2020-08-25T09:53:32.991692 #2297] DEBUG -- : lib/oxidized/input/cli.rb: Running post_login commands at RENNES_Nominal
D, [2020-08-25T09:53:32.991762 #2297] DEBUG -- : lib/oxidized/input/cli.rb: Running post_login command: "term len 0\\r", block: nil at RENNES_Nominal
D, [2020-08-25T09:53:32.991782 #2297] DEBUG -- : Telnet: term len 0\r @RENNES_Nominal
D, [2020-08-25T09:53:33.044467 #2297] DEBUG -- : lib/oxidized/model/model.rb Collecting commands' outputs
D, [2020-08-25T09:53:33.044560 #2297] DEBUG -- : lib/oxidized/model/model.rb Executing show version
D, [2020-08-25T09:53:33.044597 #2297] DEBUG -- : Telnet: show version @RENNES_Nominal
D, [2020-08-25T09:53:33.096564 #2297] DEBUG -- : lib/oxidized/model/model.rb Executing show system hardware
D, [2020-08-25T09:53:33.096641 #2297] DEBUG -- : Telnet: show system hardware @RENNES_Nominal

**Comment by cyberbohu** [comment]
what next debug lines after that line:
D, [2020-08-25T09:53:33.096641 #2297] DEBUG -- : Telnet: show system hardware @RENNES_Nominal

**Comment by sgalliani-efs** [comment]
D, [2020-08-25T11:00:16.383510 #2297] DEBUG -- : lib/oxidized/node.rb: Oxidized::Telnet failed for RENNES_Nominal
D, [2020-08-25T11:00:16.383593 #2297] DEBUG -- : lib/oxidized/job.rb: Config fetched for RENNES_Nominal at 2020-08-25 11:00:16 UTC
W, [2020-08-25T11:00:17.076982 #2297]  WARN -- : /RENNES_Nominal status no_connection, retry attempt 2
D, [2020-08-25T11:00:17.077110 #2297] DEBUG -- : lib/oxidized/worker.rb: Jobs running: 0 of 1 - ended: 0 of 6
D, [2020-08-25T11:00:17.077219 #2297] DEBUG -- : lib/oxidized/worker.rb: Added /RENNES_Nominal to the job queue
D, [2020-08-25T11:00:17.077279 #2297] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2020-08-25T11:00:17.077330 #2297] DEBUG -- : lib/oxidized/job.rb: Starting fetching process for RENNES_Nominal at 2020-08-25 11:00:17 UTC
D, [2020-08-25T11:00:17.176440 #2297] DEBUG -- : lib/oxidized/input/cli.rb: Running post_login commands at RENNES_Nominal
D, [2020-08-25T11:00:17.176518 #2297] DEBUG -- : lib/oxidized/input/cli.rb: Running post_login command: "term len 0\\r", block: nil at RENNES_Nominal
D, [2020-08-25T11:00:17.176541 #2297] DEBUG -- : Telnet: term len 0\r @RENNES_Nominal
D, [2020-08-25T11:00:17.228540 #2297] DEBUG -- : lib/oxidized/model/model.rb Collecting commands' outputs
D, [2020-08-25T11:00:17.228633 #2297] DEBUG -- : lib/oxidized/model/model.rb Executing show version
D, [2020-08-25T11:00:17.228654 #2297] DEBUG -- : Telnet: show version @RENNES_Nominal
D, [2020-08-25T11:00:17.281764 #2297] DEBUG -- : lib/oxidized/model/model.rb Executing show system hardware
D, [2020-08-25T11:00:17.281842 #2297] DEBUG -- : Telnet: show system hardware @RENNES_Nominal
D, [2020-08-25T11:00:18.077587 #2297] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2020-08-25T11:00:19.077953 #2297] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel

**Comment by sgalliani-efs** [comment]
This here is my oneos.rb file outpot : 

class OneOS < Oxidized::Model
  prompt /^([\w.@-]+[#>]\s?)$/
  comment  '! '

  # example how to handle pager
  # expect /^\s--More--\s+.*$/ do |data, re|
  #  send ' '
  #  data.sub re, ''
  # end

  # non-preferred way to handle additional PW prompt
  # expect /^[\w.]+>$/ do |data|
  #  send "enable\n"
  #  send vars(:enable) + "\n"
  #  data
  # end

  cmd :all do |cfg|
    # cfg.gsub! /\cH+\s{8}/, ''         # example how to handle pager
    # cfg.gsub! /\cH+/, ''              # example how to handle pager
    cfg.cut_both
  end

  cmd :secret do |cfg|
    cfg.gsub! /^(snmp set-read-community ").*+?(".*)$/, '\\1<secret hidden>\\2'
    cfg
  end

  cmd 'show version' do |cfg|
    comment cfg
  end

  cmd 'show system hardware' do |cfg|
    comment cfg
  end

  cmd 'show product-info-area' do |cfg|
    comment cfg
  end

  #cmd 'show running-config' do |cfg|
  cmd 'show running-config\r' do |cfg|
    cfg = cfg.each_line.to_a[0..-1].join
    cfg.gsub! /^Building configuration...\s*[^\n]*\n/, ''
    cfg.gsub! /^Current configuration :\s*[^\n]*\n/, ''
    cfg
  end

  cfg :telnet do
    username /^Username:/
    password /^Password:/
  end

  cfg :telnet, :ssh do
    # preferred way to handle additional passwords
    if vars :enable
      post_login do
        send "enable\n"
        cmd vars(:enable)
      end
    end
    post_login 'term len 0\r'
    pre_logout 'exit\r'
#    post_login 'term len 0'
#    pre_logout 'exit'
  end
end

**Comment by sgalliani-efs** [comment]
This is my oneos.rb file : 
```
_class OneOS < Oxidized::Model
  prompt /^([\w.@-]+[#>]\s?)$/
  comment  '! '

  # example how to handle pager
  # expect /^\s--More--\s+.*$/ do |data, re|
  #  send ' '
  #  data.sub re, ''
  # end

  # non-preferred way to handle additional PW prompt
  # expect /^[\w.]+>$/ do |data|
  #  send "enable\n"
  #  send vars(:enable) + "\n"
  #  data
  # end

  cmd :all do |cfg|
    # cfg.gsub! /\cH+\s{8}/, ''         # example how to handle pager
    # cfg.gsub! /\cH+/, ''              # example how to handle pager
    cfg.cut_both
  end

  cmd :secret do |cfg|
    cfg.gsub! /^(snmp set-read-community ").*+?(".*)$/, '\\1<secret hidden>\\2'
    cfg
  end

  cmd 'show version' do |cfg|
    comment cfg
  end

  cmd 'show system hardware' do |cfg|
    comment cfg
  end

  cmd 'show product-info-area' do |cfg|
    comment cfg
  end

  #cmd 'show running-config' do |cfg|
  cmd 'show running-config\r' do |cfg|
    cfg = cfg.each_line.to_a[0..-1].join
    cfg.gsub! /^Building configuration...\s*[^\n]*\n/, ''
    cfg.gsub! /^Current configuration :\s*[^\n]*\n/, ''
    cfg
  end

  cfg :telnet do
    username /^Username:/
    password /^Password:/
  end

  cfg :telnet, :ssh do
    # preferred way to handle additional passwords
    if vars :enable
      post_login do
        send "enable\n"
        cmd vars(:enable)
      end
    end
    post_login 'term len 0\r'
    pre_logout 'exit\r'
#    post_login 'term len 0'
#    pre_logout 'exit'
  end
end_

```

**Comment by cyberbohu** [comment]
You must use  \r after every cmd lines.
Like that:
cmd 'show version**\r**' do |cfg|
cmd 'show system hardware*\r*' do |cfg|
cmd 'show product-info-area*\r*' do |cfg|

**Comment by sgalliani-efs** [comment]
Sadly i'm still getting the failed telnet for some reason and it's not a network or equipement issue as i'm able to telnet to the equipement in cli.
I don't understand why oxidized is having a hard time connecting to the equipement, is it something i wrote wrong in the oneos.rb file?.

**Comment by cyberbohu** [comment]
thats mine:

```
class OneOS < Oxidized::Model
#  prompt /^([\w.@()-]+#\s?)$/
#  prompt /^([\w.@-]+[#>]\s?)$/
#  prompt /^([\w.@()-]+[#>]\s?)$/
#  prompt /^([\w.@-]+[#>]\s?)$/
#  prompt /^([\w.@-]+[#>]\s?)$/
  prompt /^([\w.@()-]+[#>]\s?)$/

  comment  '! '


  # example how to handle pager
  # expect /^\s--More--\s+.*$/ do |data, re|
  #  send ' '
  #  data.sub re, ''
  # end

  # non-preferred way to handle additional PW prompt
  # expect /^[\w.]+>$/ do |data|
  #  send "enable\n"
  #  send vars(:enable) + "\n"
  #  data
  # end

  cmd :all do |cfg|
    # cfg.gsub! /\cH+\s{8}/, ''         # example how to handle pager
    # cfg.gsub! /\cH+/, ''              # example how to handle pager
    cfg.cut_both
  end

#  cmd :secret do |cfg|
#    cfg.gsub! /^(snmp set-read-community ").*+?(".*)$/, '\\1<secret hidden>\\2'
#    cfg
#  end

  cmd "show version\r" do |cfg|
    comment cfg
  end
  cmd "show system hardware\r" do |cfg|
    comment cfg
  end

#  cmd 'show product-info-area' do |cfg|
#    comment cfg
#  end

  cmd "show running-config\r" do |cfg|
    cfg = cfg.each_line.to_a[0..-1].join
    cfg.gsub! /^Building configuration...\s*[^\n]*\n/, ''
    cfg.gsub! /^Current configuration :\s*[^\n]*\n/, ''
    cfg
  end

#  cfg :telnet do
#    username /^Username:/
#    password /^Password:/
#  end
#  cmd :secret do |cfg|
#    cfg.gsub! /^(snmp set-read-community ").*+?(".*)$/, '\\1<secret hidden>\\2'
#    cfg
#  end

  cmd "show version\r" do |cfg|
    comment cfg
  end
  cmd "show system hardware\r" do |cfg|
    comment cfg
  end

#  cmd 'show product-info-area' do |cfg|
#    comment cfg
#  end

  cmd "show running-config\r" do |cfg|
    cfg = cfg.each_line.to_a[0..-1].join
    cfg.gsub! /^Building configuration...\s*[^\n]*\n/, ''
    cfg.gsub! /^Current configuration :\s*[^\n]*\n/, ''
    cfg
  end

#  cfg :telnet do
#    username /^Username:/
#    password /^Password:/
#  end


  cfg :telnet, :ssh do
    # preferred way to handle additional passwords
    if vars :enable
      post_login do
        send "enable"
        cmd vars(:enable)
     end
    end
    post_login "term len 0\r"
    pre_logout "exit\r"
  end
end

```

**Comment by sgalliani-efs** [comment]
it's still failing at the telnet step for some reason, i'll try posting a new issue instead of this one but thank you for the help Cyber, i'll post the solution if manage to fix my problem.

**Comment by cyberbohu** [comment]
I use SSH to connect to my oneaccess. I hope you will find a solution ASAP

**Comment by sgalliani-efs** [comment]
I managed to make it work and it's backing up nicely : 

```
class OneOS < Oxidized::Model
#  prompt /^([\w.@()-]+#\s?)$/
#  prompt /^([\w.@-]+[#>]\s?)$/
#  prompt /^([\w.@()-]+[#>]\s?)$/
#  prompt /^([\w.@-]+[#>]\s?)$/
#  prompt /^([\w.@-]+[#>]\s?)$/
  prompt /^([\w.@()-]+[#>]\s?)$/

  comment  '! '

  cmd "show running-config\r" do |cfg|
    cfg = cfg.each_line.to_a[0..-1].join
    cfg.gsub! /^Building configuration...\s*[^\n]*\n/, ''
    cfg.gsub! /^Current configuration :\s*[^\n]*\n/, ''
    cfg
  end

  cfg :telnet do
    username /^Username:/
    password /^Password:/
  end

  cfg :telnet, :ssh do
    # preferred way to handle additional passwords
    if vars :enable
      post_login do
        send "enable"
        cmd vars(:enable)
     end
    end
    post_login "term len 0\r"
    pre_logout "exit\r"
  end
```

Indeed it was the SSH part that was wrong and i should have decommented the telnet line.
